### ISSUE TITLE
Loss functions

### ISSUE THREAD

**Issue by KarlNaumann** [KarlNaumann/MacroStat#37]
# Loss Functions

Implementation of a suite of common loss-functions. 

The loss-functions would take the model as input, together with a data dictionary, and maybe a dictionary of transforms, and compute the loss. 

Relevant examples would be: 
1. MSE 
2. MAPE
3. KL
4. sKL

Should check how PyTorch generally implements these loss-functions. The benefit of having these be pre-built is that we can ensure their differentiability for the autodiff approaches, so that the user just needs to generate a dictionary of comparison vectors. 

Open questions:
1. How to handle the case of comparing only a subset of indices? e.g. start and end index
